criu/cr-restore.c

```diff
@@ -2125,6 +2125,10 @@ static int restore_root_task(struct pstree_item *init)
 	__restore_switch_stage(CR_STATE_FORKING);
 
 skip_ns_bouncing:
+	pr_info("Run restore device parallel hook from criu master for external devices\n");
+	ret = run_plugins(RESTORE_DEVICE_PARALLEL);
```

Comment:
RESTORE_DEVICE_PARALLEL doesn't look like a good name. You are using it for restore device states, but other plugins can use it for other things. In this case, I think a name has to describe a point where it is called.

Reply:
This hook is called after `CR_STATE_FORKING` and before `CR_STATE_RESTORE_SIGCHLD`. Its purpose is to allow the plugin to perform certain operations in the CRIU main process in parallel with the restore operations in the target processes during `CR_STATE_RESTORE`. So could I name it `RESTORE_PARALLEL_WITH_TASKS`, or are there any other name suggestions?

Reply:
>This hook is called after CR_STATE_FORKING and before CR_STATE_RESTORE_SIGCHLD

I also agree with Andrei that it would be better for the name to describe the *place* where this plugin hook is being invoked, rather than the *purpose* of the operations performed with the AMD GPU plugin. Other plugins may use this hook for different purposes.

What about something like `PRE_RESTORE_SIGCHLD` or `POST_FORKING`?

Reply:
OK. We will use `POST_FORKING`. If there are other name suggestions, please let us know.